soh/soh/Enhancements/presets.cpp

```diff
@@ -0,0 +1,61 @@
+#include "presets.h"
+#include <variant>
+#include <string>
+#include <cstdint>
+#include <libultraship/ImGuiImpl.h>
+#include <libultraship/Cvar.h>
+#include "soh/UIWidgets.hpp"
+
+void clearCvars(std::vector<const char*> cvarsToClear) {
+    for(const char* cvar : cvarsToClear) {
+        CVar_Clear(cvar);
+    }
+}
+
+void applyPreset(std::vector<PresetEntry> entries) {
+    for(auto& [cvar, type, value] : entries) {
+        switch (type) {
+            case PRESET_ENTRY_TYPE_S32:
+                CVar_SetS32(cvar, std::get<int32_t>(value));
+                break;
+            case PRESET_ENTRY_TYPE_FLOAT:
+                CVar_SetFloat(cvar, std::get<float>(value));
+                break;
+            case PRESET_ENTRY_TYPE_STRING:
+                CVar_SetString(cvar, std::get<const char*>(value));
+                break;
+        }
+    }
+}
+
+void DrawPresetSelector(PresetType presetTypeId) {
+    const std::string presetTypeCvar = "gPreset" + std::to_string(presetTypeId);
+    const PresetTypeDefinition presetTypeDef = presetTypes.at(presetTypeId);
+    // I don't love having the hardcoded length here, but combobox needs a const and this can be a dynamic length so for
+    // the time being this should be the length of the largest amount of presets under any preset type + 1 for "default" option
+    const char* comboboxOptions[4];
```

Comment:
Finally made windows happy here, left a comment explaining the issue and the resulting hack.

Reply:
> It's a shame you squashed, but I guess you previously had: `presetTypeDef.presets.size() + 1`?

based on https://github.com/HarbourMasters/Shipwright/compare/b9bb38a21c0c3720bf97bc992e2e1cddded96576..9b133f9786df33ac5fc956e8db09d34bd7cfbc40 it looks like it indeed was
```cpp
const char* comboboxOptions[presetTypeDef.presets.size() + 1];
```

Reply:
> For Audio API we also use a dynamic list to populate it, how about using that approach here?

https://github.com/HarbourMasters/Shipwright/blob/283854c8541fc624624b8f8e603917a55c53d3a5/soh/soh/GameMenuBar.cpp#L539-L559

looks like the audio api uses `std::vector<std::pair<const char*, const char*>> GetAvailableAudioBackends() ` [from lus](https://github.com/Kenix3/libultraship/blob/85254518edcd698130a71a03e9118c512ba016be/src/menu/ImGuiImpl.cpp#L779-L781)